src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -515,5 +549,80 @@ private void UpdateStateCloseReceived()
             }
         }
         #endregion
+
+        #region Helpers
+        private static bool InitMessageWebSocketClientCertificateSupported()
+        {
+            return ApiInformation.IsPropertyPresent(
+                "Windows.Networking.Sockets.MessageWebSocketControl",
+                "ClientCertificate");
+        }
+
+        private static X509Certificate2 GetEligibleClientCertificate(X509Certificate2Collection candidateCerts)
```

Comment:
This looks like a copy/paste of the one from WInHttpCertificateHelper with an if 0 => null changed to an assert.  Can they share?

They both seem to lack the `if Count == 1 => use it` optimization.